content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,537 @@
+ 
+The Chi-square test helps in determining if there is a notable difference between the observed and normal frequencies in one or more categories, the values of petal_width must be changed into categories of zeros and ones using the 50% percentile which is the second quartile as the yardstick.
+ 
+There are clear [differences](https://byjus.com/maths/difference-between-percentage-and-percentile/) between percentage and percentile. As percentage is just a fraction of a hundred, the percentile is more concerned with ranks, it can be termed as the percentage of values that fall below a certain value mark.
+ 
+It is rather used to find the position of a value in a given set of values. For example, if a girl has the 50th percentile on an examination of 200, we can say that by scoring 200 marks, she has higher marks than 50% of the remaining class.
+There are two conditions that must be satisfied while calculating percentile.
+1. Data must be arranged in ascending order.
+2. if the index is not an integer, it must be rounded up.
+```python
+print(np.percentile(df.petal_width, 50))
+```
+```bash
+1.3
+```
+
+A quartile is another term related to percentile, it is a specified percentile, we have 1st quartile as 25% percentile, second quartile as 50%, and 3rd as 75% percentile. You can as well read this [article](https://datascienceupskill.com/statistics/percentiles-and-quartiles) to get a better grip. 
+ 
+```python
+def petal_cat(df):
+ if df['petal_width'] <= 1.3:
+   return 0
+ elif df['petal_width'] > 1.3:
+   return 1
+ else:
+   return 'indifferent'
+ 
+df['petal_width_new'] = df.apply(petal_cat, axis=1) 
+```
+The non-machine-friendly format for our *species* class has to be transformed to 0,1 and 2, which is a machine-readable format for the testing to be carried out. Therefore, with [label-encoding](/engineering-education/content/articles/introduction-to-scikit-learn-in-python/
+), a numeric value will be assigned to each of the categories in the class. To learn more about label encoding, [click](/engineering-education/content/articles/data-preprocessing-in-r/).
+ 
+```python
+def species_cat(df):
+ if df["species"] == "Iris-virginica":
+   return 0
+ elif df["species"] == "Iris-versicolor":
+   return 1
+ else:
+   return 2
+ 
+df["species"] = df.apply(species_cat, axis=1)
+```
+```python
+df.sample(5)
+```
+![dfsample5](/engineering-education/statistical-hypothesis-testing-python-implementation/dfsample5.PNG)
+```python
+df_new = df.drop(columns=["sepal_width", "sepal_length", "petal_length", "petal_width"])
+```
+```python
+df_new.head()
+```
+![newhead](/engineering-education/statistical-hypothesis-testing-python-implementation/df_new.PNG)
+ 
+Let's make use of `chic2_contigency` provided for us in the `scipy.stats` library.
+```python
+stat, p, dof, expected= chi2_contingency(df_new)
+ 
+print(dof)
+```
+```bash
+The degree of freedom is: 149
+```
```

Comment:
The print statement does not match with the output. Please edit the print statement appropriately so that when the reader tries to run this piece of code, the output matches.